cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableDataLogRepairCommand.java

```diff
@@ -0,0 +1,914 @@
+import lombok.Cleanup;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.val;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiConsumer;
+
+/**
+ * This command provides an administrator with the basic primitives to manipulate a DurableLog with damaged entries.
+ * The workflow of this command is as follows:
+ * 1. Disable the original DurableLog (if not done yet).
+ * 2. Reads the original damaged DurableLog and creates a backup copy of it for safety reasons.
+ * 3. Validate and buffer all the edits from admin to be done on the original DurableLog data (i.e., skip, delete,
+ * replace operations). All these changes are then written on a Repair Log (i.e., original DurableLog data + admin changes).
+ * 4. With the desired updates written in the Repair Log, the admin can replace the original DurableLog metadata by the
+ * Repair Log's one. This will make the DurableLog for the Segment Container under repair to point to the Repair Log data.
+ * 5. The backed-up data for the originally damaged DurableLog can be reused to create a new Repair Log or discarded if
+ * the Segment Container recovers as expected.
+ */
+public class DurableDataLogRepairCommand extends DataRecoveryCommand {
+
+    private final static Duration TIMEOUT = Duration.ofSeconds(10);
+
+    /**
+     * Creates a new instance of the DurableLogRepairCommand class.
+     *
+     * @param args The arguments for the command.
+     */
+    public DurableDataLogRepairCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(1);
+        int containerId = getIntArg(0);
+        val bkConfig = getCommandArgs().getState().getConfigBuilder()
+                .include(BookKeeperConfig.builder().with(BookKeeperConfig.ZK_ADDRESS, getServiceConfig().getZkURL()))
+                .build().getConfig(BookKeeperConfig::builder);
+        @Cleanup
+        val zkClient = createZKClient();
+        @Cleanup
+        val dataLogFactory = new BookKeeperLogFactory(bkConfig, zkClient, this.executorService);
+        dataLogFactory.initialize();
+
+        // Open the Original Log in read-only mode.
+        @Cleanup
+        val originalDataLog = dataLogFactory.createDebugLogWrapper(containerId);
+
+        // Disable the Original Log, if not disabled already.
+        output("Original DurableLog is enabled. You are about to disable it.");
```

Comment:
Is it possible to tell the user that the log is disabled and we can proceed further instead of trying to disable an already disabled log?